codec-socks/src/main/java/io/netty/handler/codec/socks/SocksCmdResponseDecoder.java

```diff
@@ -28,65 +28,57 @@
  */
 public class SocksCmdResponseDecoder extends ReplayingDecoder<State> {
 
-    private SocksProtocolVersion version;
-    private int fieldLength;
-    private SocksCmdStatus cmdStatus;
-    private SocksAddressType addressType;
-    private byte reserved;
-    private String host;
-    private int port;
-    private SocksResponse msg = SocksCommonUtils.UNKNOWN_SOCKS_RESPONSE;
-
     public SocksCmdResponseDecoder() {
         super(State.CHECK_PROTOCOL_VERSION);
     }
 
     @Override
     protected void decode(ChannelHandlerContext ctx, ByteBuf byteBuf, List<Object> out) throws Exception {
+        SocksCmdStatus cmdStatus = null;
+        SocksAddressType addressType = null;
         switch (state()) {
             case CHECK_PROTOCOL_VERSION: {
-                version = SocksProtocolVersion.valueOf(byteBuf.readByte());
+                SocksProtocolVersion version = SocksProtocolVersion.valueOf(byteBuf.readByte());
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Add a default case to this switch. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3ASwitchLastCaseIsDefaultCheck)